intel_extension_for_transformers/llm/library/jblas/jblas/kernel_jit.h

```diff
@@ -740,6 +743,430 @@ static inline JBLAS_CODE decompress_s4_s8(utils::int4x2* srcptr, int8_t* dstptr,
   DecompressS4S8_AVX512F::forward(srcptr, dstptr, (size_t)row * col);
   return JblasSuccess;
 }
+
+// src: row x col => dst: ⌈col/n_tile⌉ x ⌈row/row_pack⌉ x n_tile x row_pack (zeor-padded)
+// Extra padding can be applied with memset calls in `static void forward(...)`
+class PaddingInterleaveCvt : protected xbyak::JitAvx512f {
+ public:
+  struct params {
+    const void* srcptr;
+    void* dstptr;
+    int row, col;
+    int srcstride, dststride;  // dst = dst_base + dststride * n_idx, where n_idx % n_tile == 0
+  };
+  typedef void (*func_t)(params* p);
+  void operator()(params* p) const { mKernel(p); }
+
+ private:
+  static inline const uint16_t idx_interleave_self[32] = {
+      0,  16, 1,  17, 2,  18, 3,  19,  //
+      4,  20, 5,  21, 6,  22, 7,  23,  //
+      8,  24, 9,  25, 10, 26, 11, 27,  //
+      12, 28, 13, 29, 14, 30, 15, 31,  //
+  };
+
+  PaddingInterleaveCvt(int n_tile, JBLAS_DTYPE dst_t) : PaddingInterleaveCvt(n_tile, dst_t, dst_t) {}
+  PaddingInterleaveCvt(int n_tile, JBLAS_DTYPE dst_t, JBLAS_DTYPE src_t, int row_pack = 0) : xbyak::JitAvx512f() {
+    inLocalLabel();  // use local label for multiple instance
+    const auto src_bytes = static_cast<int>(utils::jblas_dtype_size(src_t));
+    const auto dst_bytes = static_cast<int>(utils::jblas_dtype_size(dst_t));
+    if (row_pack == 0) row_pack = 4 / dst_bytes;  // default value
+    const auto ne_zmm = 64 / std::max(src_bytes, dst_bytes);
+    const auto src_bytes_vmm = ne_zmm * src_bytes;
+    const auto dst_bytes_vmm = ne_zmm * dst_bytes;
+
+    assert(n_tile % ne_zmm == 0);
```

Comment:
should be ```static_assert```?
throw some comments about why fail

Reply:
I tried to avoid the use of template in code generators to potentially save some time for compilation. It should be harmless to add some runtime branches in the logic of JIT code generation. As a result, it is not a constexpr here.

Reply:
ok, for these asserts, please throw some explanation, just for example
```suggestion
    assert(n_tile % ne_zmm == 0, "n_tile must be divided by (64/sizeof(dt))");
```

Reply:
Unlike `static_assert`, assert marco function [does not have a second parameter](https://en.cppreference.com/w/cpp/error/assert). But yes, we can try to use the workaround introduced in the link.

However, I would like to leave it as it is for now until next time when I need to update jblas to avoid jblas syncing conflictions.  (and I'll remember to invite you to review).